zerver/tests/test_openapi.py

```diff
@@ -309,21 +382,24 @@ def test_openapi_arguments(self) -> None:
         for p in urlconf.v1_api_and_json_patterns:
             if p.lookup_str != 'zerver.lib.rest.rest_dispatch':
                 continue
+
+            # since the module was already imported and is now residing in
+            # memory, we won't actually face any performance penalties here.
             for method, value in p.default_args.items():
                 if isinstance(value, str):
-                    function = value
+                    function_name = value
                     tags = set()  # type: Set[str]
                 else:
-                    function, tags = value
+                    function_name, tags = value
+
+                lookup_parts = function_name.split('.')
+                module = __import__('.'.join(lookup_parts[:-1]), {}, {}, [''])
```

Comment:
Does this eliminate the need to have those imports at the top of the class?  If so, might make sense to extract a prep commit that replaces those imports with this.

Reply:
@timabbott: Ah, no, this does not. In fact this import isn't even an import (I think), what we're doing is just retrieving the function from memory.

So my logic here was that by importing urls.py, since we've already imported every view function (which is what triggers all of the has_request_variables decorators), all of the functions are already loaded into runtime memory. Thus with this "import" we're just getting a handle to that function.

Reply:
I've explained more in the (rather lengthy) commit message.

Reply:
You're correct that if we've already imported a module, this will just look it up and not import it again.  However, if we hadn't, this line will import it.  Which is why I think this should be able to allow us to remove the views imports for views not always imported during Django setup, above.

Reply:
You're correct, we can remove those lines. Good observation! Thank you for the suggestion!